.github/workflows/build.yaml

```
@@ -1,26 +1,363 @@
 name: "Test"
 
 on:
-    pull_request:
+  pull_request:
+  push:
+    branches:
+      - master
+    tags:
+      - "v*"
 
 jobs:
-    build:
-        name: "MacOS build"
+    tarball:
+      runs-on: ubuntu-latest
+      name: Tarball
+      steps:
+        - name: Clone wake
+          uses: actions/checkout@v3
+          with:
+            fetch-depth: 0
+
+        - name: Install Deps
+          run: sudo apt-get update && sudo apt-get install -y build-essential fuse libfuse-dev libsqlite3-dev libgmp-dev libncurses5-dev pkg-config git g++ gcc libre2-dev python3-sphinx clang-format-12
+
+        - name: Check Formatting
+          run: clang-format-12 --style=file --Werror -n $(./scripts/which_clang_files all)
+
+        - name: Build Tarball
+          run: make tarball
+
+        - name: Run Tests
+          run: make test && make unittest
+
+        - name: Generate Docs
+          run: mkdir www && ./bin/wake --no-workspace --html > www/index.html
+
+        - name: wake to rst
+          run: PATH=$(pwd)/bin:$PATH PYTHONPATH=$(pwd)/scripts python3 scripts/wake2rst.py
+
+        - name: Upload tarball
+          uses: actions/upload-artifact@v3
+          with:
+            name: tarball
+            path: wake_*.tar.xz
+
+        - name: Upload Dockerfiles
+          uses: actions/upload-artifact@v3
+          with:
+            name: dockerfiles
+            path: .circleci/dockerfiles/
+            retention-days: 1
+
+        - name: Upload debian files
+          uses: actions/upload-artifact@v3
+          with:
+            name: debian
+            path: debian
+            retention-days: 1
+
+        - name: Upload test artifacts
+          uses: actions/upload-artifact@v3
+          with:
+            name: tests
+            path: tests
+            retention-days: 1
+
+        - name: Upload index.html
+          uses: actions/upload-artifact@v3
+          with:
+            name: index.html
+            path: www/index.html
+            retention-days: 1
+
+        - name: Upload html
+          uses: actions/upload-artifact@v3
+          with:
+            name: html
+            path: scripts/sphinx/build/html
+            retention-days: 1
+
+    docker_builds:
+      runs-on: ubuntu-latest
+      needs: tarball
+      name: Docker Build (${{ matrix.target }})
+      strategy:
+        matrix:
+          include:
+            - target: alpine
+              dockerfile: alpine
+              extra_docker_build_args:
+              extra_docker_run_args:
```

The syntax highlighter is complaining about these being empty; I imagine the keys are in here because you want them here, but would explicit empty strings break anything?  (It fixes the highlighter, but I'm not sure about the Workflows parser.)
```suggestion
              extra_docker_build_args: ""
              extra_docker_run_args: ""
```